client/src/app/site/deployment-center/deployment-center-setup/wizard-logic/deployment-center-state-manager.ts

```diff
@@ -463,13 +463,32 @@ export class DeploymentCenterStateManager implements OnDestroy {
     this._localStorageService.setItem(storedPermissionItem.id, storedPermissionItem);
   }
 
-  public getVstsPassthroughHeaders(appendMsaPassthroughHeader: boolean = false): Headers {
-    const headers = new Headers();
-    headers.append('Content-Type', 'application/json');
-    headers.append('Accept', 'application/json');
-    headers.append('Vstsauthorization', `Bearer ${this._vstsApiToken}`);
-    headers.append('MsaPassthrough', `${appendMsaPassthroughHeader}`);
-    return headers;
+  private _getVstsToken(): Observable<{ status: string; statusMessage: string; result: any }> {
+    return this._portalService
+      .getAdToken('azureTfsApi')
+      .first()
+      .switchMap(tokenData => {
+        if (tokenData == null || tokenData.result == null || tokenData.result.token == null || !parseToken(tokenData.result.token)) {
+          return Observable.of({
+            status: 'failed',
+            statusMessage: this._translateService.instant(PortalResources.vstsTokenIsInvalid),
+            result: null,
+          });
+        } else {
+          return Observable.of({
+            status: 'succeeded',
```

Comment:
We could use enums for status

Reply:
Yes enum makes more sense. But to make this change, i will have make changes in Kudu deployment code as well. Hence will not do this.